<launch>
    <arg name="show" default="false" />
    <arg name="output" default="screen" />
    <arg name="manager"     default="swarm_manager"/>
    <arg name="send_img"     default="false"/>
    <arg name="send_whole_img_desc"     default="false"/>
    <arg name="self_id"     default="1"/>
    <arg name="camera_config_path" default="/root/swarm_ws/src/VINS-Fusion-Fisheye/config/fisheye_ptgrey_n3/front.yaml" />
    <node pkg="nodelet" type="nodelet" name="swarm_loop" 
        args="load swarm_localization_pkg/SwarmLoopNode /$(arg manager) --no-bond" output="screen" >
        <rosparam>
            lcm_uri: udpm://224.0.0.251:7667?ttl=1
            loop_image_downsample: 1.0
            init_loop_min_feature_num: 8
            init_loop_min_feature_num_l2: 8
            min_loop_feature_num: 12
            jpg_quality: 75
            
            query_thres: 100
            init_query_thres: 100

            min_movement_keyframe: 0.1
            enable_pub_remote_img: false
            enable_sub_remote_img: false
            max_freq: 0.5
            accept_min_3d_pts: 30
            match_index_dist: 5
            recv_msg_duration: 0.5
            superpoint_thres: 0.02
            triangle_thres: 0.012
            #triangle_thres: 0.008
            min_direction_loop: 3
            is_pc_replay: false
            detector_match_thres: 0.7
            output_path: /root/output/
            send_all_features: false
            lower_cam_as_main: false
            min_match_per_dir: 5
        </rosparam>
        <param name="superpoint_model_path" value="$(find swarm_loop)/models/superpoint_v1_tx2_fp16.trt" type="string" />
        <param name="netvlad_model_path" value="$(find swarm_loop)/models/mobilenetvlad_208x400_tx2_fp16.trt" type="string" />
       <param name="debug_image" value="$(arg show)" type="bool" />
       <param name="send_img" value="$(arg send_img)" type="bool" />
       <param name="send_whole_img_desc" value="$(arg send_whole_img_desc)" type="bool" />
       <param name="self_id" value="$(arg self_id)" type="int" />
       <param name="camera_config_path" value="$(arg camera_config_path)" type="string" />
    </node>
</launch>
